File: src/components/externalLink.js
import React from "react"
import PropTypes from "prop-types"

import { FontAwesomeIcon } from "@fortawesome/react-fontawesome"

const ExternalLink = ({ text, icon, src }) => (
  <a href={src} target="_blank" rel="noopener noreferrer">
    <div className="externalLink">
      <div className="externalLinkText">/{text}/</div>
      <div className="externalLinkIcon">
        <FontAwesomeIcon icon={icon} />
      </div>
    </div>
  </a>
)

ExternalLink.defaultProps = {
  text: "Link",
  src: "",
}

ExternalLink.propTypes = {
  text: PropTypes.string,
  icon: PropTypes.object.isRequired,
  src: PropTypes.string,
}

export default ExternalLink

File: src/components/external_link.rs
use yew::{Html, Properties, classes, component, html};

#[derive(Properties, PartialEq)]
pub struct ExternalLinkProps {
  pub href: String,
  pub text: String,
  pub icon: Html,
}

#[component]
pub fn ExternalLink(props: &ExternalLinkProps) -> Html {
  let text = format!("/{}/", props.text);

  html! {
    <a href={props.href.clone()} rel="noopener noreferrer" target="_blank" class={classes!("external-link")}>
      <div class={classes!("external-link__text")}>{text}</div>
      <div class={classes!("external-link__icon")}>{props.icon.clone()}</div>
    </a>
  }
}
